geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/CommandManager.java

```diff
@@ -127,15 +134,17 @@ private static void raiseExceptionIfEmpty(Set<Class<?>> foundClasses, String err
     return userCommandPackages;
   }
 
-  private void loadUserCommands(ClasspathScanLoadHelper scanner, Set<String> restrictedToPackages) {
-    if (restrictedToPackages.size() == 0) {
+  private void loadUserDefinedCommands() {
+    String[] userCommandPackages = getUserCommandPackages().toArray(new String[] {});
+
+    if (userCommandPackages.length == 0) {
       return;
     }
 
     // Load commands found in all of the packages
-    try {
-      Set<Class<?>> foundClasses = scanner.scanPackagesForClassesImplementing(CommandMarker.class,
-          restrictedToPackages.toArray(new String[] {}));
+    try (ClasspathScanLoadHelper scanner = new ClasspathScanLoadHelper(userCommandPackages)) {
+      Set<Class<?>> foundClasses =
+          scanner.scanPackagesForClassesImplementing(CommandMarker.class, userCommandPackages);
```

Comment:
Not sure if there will be any perf benefits, just thought of bringing it here...It looks like we are instantiating the command classed found (below line); can this initialization be done lazily, when the command is used/executed, will this help.

Reply:
Yes, you are correct that the preemptive initialization of the command, without actually using it, would be slightly faster. But also, given that the number of "userCommandPackages" are unknown here and most likely never going to be populated is a safe bet to do this. In addition I don't think this feature has been documented (but I might be wrong, but I'm not going to go through 1000+ pages of documentation to prove it right/wrong), I doubt this path is ever used. BUT I am keeping it here, just in case a user might actually be using this functionality.